app/models/user.rb

```diff
@@ -318,6 +318,13 @@ def set_mhv_ids(mhv_id)
 
   # Other MPI
 
+  def validate_mpi_profile
+    return unless mpi_profile?
+
+    raise SignIn::Errors::MPILockedAccountError.new message: 'Death Flag Detected' if mpi_profile.deceased_date
```

Comment:
Could we have an error that doesn't live in the `SignIn` directory? I'm not super sure the best place for the error, maybe in `mpi/errors.rb`? It would just be nice to keep the `SignIn` stuff completely separate from the rest of the code

Reply:
I'm creating a new 'AccountLocked` error within the `lib/mpi/errors/errors.rb` file for this.